hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -268,7 +269,25 @@ case class HoodieResolveReferences(sparkSession: SparkSession) extends Rule[Logi
       } else {
         l
       }
-
+    // Fill schema for Create Table without specify schema info
+    case c @ CreateTable(tableDesc, _, _)
+      if isHoodieTable(tableDesc) =>
+        val tablePath = getTableLocation(c.tableDesc, sparkSession)
+          .getOrElse(s"Missing location defined in table ${c.tableDesc.identifier}")
+        val metaClient = HoodieTableMetaClient.builder()
+          .setBasePath(tablePath)
+          .setConf(sparkSession.sessionState.newHadoopConf())
+          .build()
+        val tableSchema = HoodieSqlUtils.getTableSqlSchema(metaClient).map(HoodieSqlUtils.addMetaFields)
+        if (tableSchema.isDefined && tableDesc.schema.isEmpty) {
+          // Fill the schema with the schema from the table
+          c.copy(tableDesc.copy(schema = tableSchema.get))
+        } else if (tableSchema.isDefined && tableDesc.schema != tableSchema.get) {
```

Comment:
should we do "!=" or ".equals()" for schema comparison?

Reply:
well, In scala, we usually use "=" or "!=" test compile two object.